src/Security/Authorization/Core/src/AuthorizationOptions.cs

```diff
@@ -11,7 +12,10 @@ namespace Microsoft.AspNetCore.Authorization;
 /// </summary>
 public class AuthorizationOptions
 {
+    private static readonly Task<AuthorizationPolicy?> _nullPolicyTask = Task.FromResult<AuthorizationPolicy?>(null);
+
     private Dictionary<string, AuthorizationPolicy> PolicyMap { get; } = new Dictionary<string, AuthorizationPolicy>(StringComparer.OrdinalIgnoreCase);
+    private Dictionary<string, Task<AuthorizationPolicy?>> PolicyTaskMap { get; } = new Dictionary<string, Task<AuthorizationPolicy?>>(StringComparer.OrdinalIgnoreCase);
```

Comment:
Why keep both?

Reply:
Aha! `PolicyTaskMap` alone is enough! I removed the other one.